### DISCUSSION TITLE
azure-cognitiveservices-speech

### DISCUSSION THREAD

**Discussion by mxav1111** [flet-dev/flet#4464] [Packages] [answered]
Having issues with flet build :-
  $ flet build web -vvv --module-name mod1  --output ../app1 
## ERROR: No matching distribution found for azure-cognitiveservices-speech. 

pip install and uninstall works fine with azure-cognitiveservices-speech

**Comment by ndonkoHenri** [comment]
Converting this discussion from General to Packages.

Package: https://pypi.org/project/azure-cognitiveservices-speech

**Comment by mxav1111** [comment]
Thank you.  Is there anything i can do,?please let me know.  I can try. 🙏

**Comment by FeodorFitsner** [comment]
It's not even an open source: https://pypi.org/project/azure-cognitiveservices-speech/#files

**Reply by mxav1111** [reply]
>>> It's not even an open source: https://pypi.org/project/azure-cognitiveservices-speech/#files

I can try requesting assistance from Microsoft Support through formal channel if that helps (They have access to product teams).
If you could guide me and propose a solution,  we will create the ticket and provide our suggestion too while updating them about this issue.

**Reply by FeodorFitsner** [reply]
you think they'll be willing to build and publish a package for Android/iOS? You can give it a try.
It could be done with https://github.com/flet-dev/mobile-forge we don't have instructions for users yet, but if they decide to do that I can provide assistance.

**Reply by mxav1111** [reply]
>>> you think they'll be willing to build and publish a package for Android/iOS? You can give it a try. It could be done with https://github.com/flet-dev/mobile-forge we don't have instructions for users yet, but if they decide to do that I can provide assistance.

But failing command line is  flet build web ....

**Reply by FeodorFitsner** [reply]
Then they need to build pyodide (used to run python in the browser) package: https://pyodide.org/en/stable/development/new-packages.html

**Comment by Muddassir-Farooq-official** [comment]
> It's not even an open source: https://pypi.org/project/azure-cognitiveservices-speech/#files

@FeodorFitsner there is a serious issue kindly resolve it I. Mention many time but no response aab file are not signing correctly and if it happens then it displays blank screen when download from Google Play store

**Comment by ndonkoHenri** [comment, marked as answer]
To wrap this up: as Feodor noted, this package can't be packaged for Flet.

- The Python package is a thin ctypes wrapper over a closed-source, proprietary native library (`libMicrosoft.CognitiveServices.Speech.core`). There's no source distribution on PyPI, so there's nothing to cross-compile — and the bundled binaries are desktop-only, not mobile.
- Microsoft ships mobile only as a separate Java AAR / Swift xcframework, which this Python wrapper can't load.

However, Azure Speech has a full REST API for both speech-to-text and text-to-speech. Calling it over plain HTTP (e.g. `requests`/`httpx`) works from pure Python on mobile with no native wheel — this could be a better approach for a Flet app. If you need streaming/real-time recognition specifically, the native SDK via [pyjnius/pyobjus](https://flet.dev/blog/tap-into-native-android-and-ios-apis-with-Pyjnius-and-pyobjus) could be a possibility.
